DatadogInternal/Sources/NetworkInstrumentation/URLSession/URLSessionSwizzler.swift

```diff
@@ -24,8 +24,14 @@ internal final class URLSessionSwizzler {
         defer { lock.unlock() }
         dataTaskURLRequestCompletionHandler = try DataTaskURLRequestCompletionHandler.build()
         dataTaskURLRequestCompletionHandler?.swizzle(interceptCompletion: interceptCompletionHandler)
-        dataTaskURLCompletionHandler = try DataTaskURLCompletionHandler.build()
-        dataTaskURLCompletionHandler?.swizzle(interceptCompletion: interceptCompletionHandler)
+
+        if #available(iOS 13.0, *) {
+            // Prior to iOS 13.0 the `URLSession.dataTask(with:url, completionHandler:handler)` makes an internal
+            // call to `URLSession.dataTask(with:request, completionHandler:handler)`. To avoid duplicated call
+            // to the callback, we don't apply below swizzling prior to iOS 13.
+            dataTaskURLCompletionHandler = try DataTaskURLCompletionHandler.build()
+            dataTaskURLCompletionHandler?.swizzle(interceptCompletion: interceptCompletionHandler)
+        }
```

Comment:
This change is porting the configuration from previous version of URLSession instrumentation, this: https://github.com/DataDog/dd-sdk-ios/blob/506d6625dde90205f6f17c5e8369ed978c09fdd5/DatadogInternal/Sources/NetworkInstrumentation/URLSession/URLSessionSwizzler.swift#L34-L40 

This is required to make unit tests pass on iOS 12.4. Without this change, [the `expectation` in this test](https://github.com/DataDog/dd-sdk-ios/blob/2e5e4c4e19f6205535779920d3c593fc2813eab1/DatadogInternal/Tests/NetworkInstrumentation/URLSessionSwizzlerTests.swift#L20) was fulfilled 4 times instead of 2. 

Rationale: The promise of `URLSessionSwizzler` is to intercept completion handler by abstracting runtime details (swizzling). In no circumstances this block should be called more than once. If iOS 12.4 details is different, the difference should be erased in this abstraction, delivering the right promise to higher level.